source/game.programme.programmelicence.bmx

```diff
@@ -758,7 +758,15 @@ Type TProgrammeLicence Extends TBroadcastMaterialSource {_exposeToLua="selected"
 
 	Method isLive:int() {_exposeToLua}
 		if GetSubLicenceCount() = 0 and GetData()
+			'TODO this is not 100% correct, yet
+			'a programme with a fixed live time will still be shown as live
+			'as long as it is not broadcasted...,
 			return data.IsLive() And GetTimesBroadcasted(owner) < 1
+			'alternative?
+			'if GetTimesBroadcasted(owner) >= 1 then return false
+			'if data.IsAlwaysLive() then return true
```

Comment:
GetTimesBroadcasted(owner)-check: wenn ein "zurueckgesetztes" ("wieder live"-gemachtes) Programm erneut vom Haendler gekauft wird, ist "GetTimesBroadcasted(owner)" bereits > 0, wenn der Spieler es in der Vergangenheit bereits ausgestrahlt haben sollte.

Reply:
Und wenn irgendwie die TimesBroadcasted genutzt wuerden, waere es denke ich uninteressant, ob der Besitzer ("owner") es bereits ausstrahlte - wichtiger ist, ob irgendwer ("-1") es schon jemals ausstrahlte.

Vielleicht kann ein "if data.IsLiveOnTape() then Return False" als Ersatz dienen?

Reply:
Weiterer Gedanke: Wenn der "Live"-Flag nur noch ein Indikator dafuer ist, dass es sich um ein Programm mit "Live-Element" handelt, dann muss "LiveOnTape" anders interpretiert werden.

"LiveOnTape" bedeutet ja dann auch nur, dass es sich um ein ehemaliges "Live-Element" handelt ("Live-Ereignis beendet und auf Video gebannt").

Koennte man da nicht generell "LiveOnTape" auch umbenennen in "ProductionFinished" oder "RecordingDone" ... Sprich einen Flag der nur anzeigt, dass keine Aufnahmen mehr stattfinden / Produktion abgeschlossen ist.

"LiveOnTape" waere dann entsprechend nur noch die Kombination der Flags "Live + ProductionFinished".

Und mittels des neuen Flags "ProductionFinished" liese sich obiges "isLive" ja einfach umsetzen - anstatt ueber "GetTimesBroadcasted" etc gehen zu muessen.

Reply:
GetTimesBroadcasted-Check: Der aktuell von mir verfolgte Ansatz ist ja, bei Live-Programmen, die das RefillLimit-Flag gesetzt haben, die Anzahl der Ausstrahlungen auf 0 zurückzusetzen. Ein zurückgegebenes/entferntes Programm ist bei Wiedererwerb eben nicht das schon einmal gesendete sondern ein "frisches".

Das LiveOnTape-Flag verliert dabei etwas an seiner Bedeutung. Ich versuche aktuell, die beiden Flags als Konfiguration (nicht änderbar) zu interpretieren, dafür aber den Getter-Aufrufen mehr Bedeutung zu geben (die reagieren dynamisch auf die aktuelle Situation). Ich bin noch nicht 100%ig sicher, ob das klappt.
Vielleich schwenke ich auch nochmal auf das Ändern der Flags bei Ausstrahlung etc. um. Dann muss man sich für das "Refill" eines Live-Programms aber merken, dass es mal live war. Die Ausstrahlungshäufigkeit müsste dann trotzdem auf 0 gesetzt werden, da ja das "neue" Live-Programm noch nicht ausgestrahlt worden sein kann.